Extending/Backoffice-Search/index.md

```diff
@@ -0,0 +1,84 @@
+---
+keywords: Backoffice Search
+versionFrom: 8.6.0
+meta.Title: "Backoffice Search"
+meta.Description: "A guide to customization of Backoffice Search"
+---
+
+# Backoffice Search
+
+Backoffice search is usefull tool, which support you in finding your content. Umbraco by default will allow you to search using only few fields: 
+
+| Node Type    | Propagated Fields    |
+| ------------ | -------------------- |
+| All Nodes    | UmbracoFileFieldName |
+| Media Nodes  | File path            |
+| Member Nodes | email, loginName     |
+|              |                      |
+
+Umbraco 8.6 introduce new easy way for extend/override that fields, which is called IUmbracoTreeSearcherFields. This service exposed hardcoded internal list of fields and allow for modification on them.
+
+### Adding custom properties to backoffice search
+
+#### All Node types
+
+```
+public class CustomInternalSearchConstants : InternalSearchConstants,     IInternalSearchConstants
+{
+    public List<string> GetBackOfficeFields()
+
+    {
+        var list = base.GetBackOfficeFields();
+        list.Add("parentID");
```

Comment:
I think with the examples we should provide more real worldy things, such as in the suggested text above, eg searching Members by an Organisation or Company Name property - searching content by BodyText alias?

Reply:
@marcemarc here I was using fields which are in all solution and are not related to any on specific starter kits/demo sites

Reply:
ahh I see, to ensure the example works for everyone - I was thinking it would be cool to show the possibilities though - perhaps with a comment above explaining that the property being added was a custom one, whereas I couldn't think why you would want to search by ParentId?, so would it confuse people as to what the purpose of the example was for... maybe creatorName or writerName, or __NodeTypeAlias would be useful? but difficult, because you want it to be useful text that might contain keywords to be searched... so maybe showing custom properties being used isn't so bad...

Reply:
@marcemarc when I was implementing that tool my client request possibility of searching by parentID :)

Reply:
hehe

Reply:
Think 'we could' have one other different example to parentid? - I don't mind adding it in after we merge... it would be good to show the difference between GetBackOfficeFields and the other specific three Getters, in that if you wanted to add ParentId to all three, you'd only need to add it to the GetBackOfficeFields one...?

Also I think we should add an example of registering the new CustomUmbracoTreeSearcherFields in a composer, so people can see they need to do that step too.

Reply:
@marcemarc 
 >>if you wanted to add ParentId to all three, you'd only need to add it to the GetBackOfficeFields one...?
Yes you need only add to GetBackOfficeFields
about registering composer, hmm I have mixed feeling, but yeah probably a good idea as someone could be not familiar with Umbraco

Reply:
Yep, I think I feel the same thing, eg not repeating everything everywhere, but I don't think people will necessarily know to use RegisterUnique

Reply:
Probably we should get input from doc team @sofietoft, can you help us here or point to a person which should be involved:)?

Reply:
@bielu we are in the Documentation meeting now, so I am discussing with the rest of the team!

Reply:
@marcemarc cool :)